test/metabase/db/custom_migrations_test.clj

```diff
@@ -1014,3 +1020,445 @@
+                                                                           :creator_id user-id
+                                                                           :parameters []})
+                [dashcard-id]  (t2/insert-returning-pks! :model/DashboardCard {:dashboard_id           dashboard-id
+                                                                               :visualization_settings dashcard-vis
+                                                                               :card_id                card-id
+                                                                               :size_x                 4
+                                                                               :size_y                 4
+                                                                               :col                    1
+                                                                               :row                    1})]
+            (let [expected-settings {:graph.dimensions ["CREATED_AT" "CATEGORY"],
+                                     :graph.metrics    ["count"],
+                                     :click            "link",
+                                     :click_link_template
+                                     "http://localhost:3001/?year={{CREATED_AT}}&cat={{CATEGORY}}&count={{count}}"
+                                     :click_behavior
+                                     {:type         "link",
+                                      :linkType     "url",
+                                      :linkTemplate "http://localhost:3001/?year={{CREATED_AT}}&cat={{CATEGORY}}&count={{count}}"},
+                                     :column_settings
+                                     ;; the model keywordizes the json parsing yielding this monstrosity below
+                                     {"[\"ref\",[\"field\",2,null]]"
+                                      {:click_behavior
+                                       {:type             "link",
+                                        :linkType         "url",
+                                        :linkTemplate     "http://example.com/{{ID}}",
+                                        :linkTextTemplate "here's an id: {{ID}}"}},
+                                      "[\"ref\",[\"field\",6,null]]"
+                                      {:click_behavior
+                                       {:type             "link",
+                                        :linkType         "url",
+                                        :linkTemplate     "http://example.com//{{id}}",
+                                        :linkTextTemplate "here is my id: {{id}}"}}}}]
+              (f)
+              (is (= expected-settings
+                     (-> (t2/select-one DashboardCard :id dashcard-id)
+                         :visualization_settings))))))]
+    (testing "Running the migration from scratch"
```

Comment:
could you add a test case for to check if the data migration has been ran previously, we don't run this migration?

same for the other migration.